openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -0,0 +1,91 @@
+"""Signal handlers for writing course dates into edx_when."""
+from __future__ import absolute_import, unicode_literals
+
+import logging
+
+from django.dispatch import receiver
+from six import text_type
+from xblock.fields import Scope
+from xmodule.modulestore.django import SignalHandler, modulestore
+
+from edx_when.api import FIELDS_TO_EXTRACT, set_dates_for_course
+
+log = logging.getLogger(__name__)
+
+
+def date_field_values(date_fields, xblock):
```

Comment:
Nothing about this method is date-specific. It just takes settings-field names and returns values, if they exist. Do we not have that elsewhere? If not, maybe we should move this more centrally?

Reply:
Good question, I'll look.

Reply:
I don't think this particular function is that useful as a general utility method. However, I do think renaming it, and marking it as internal, is worth doing.